CMS-Enterprise/mint-app
# [MINT-3228] Edit milestone modal

# MINT-3228

## Description


- Added `EditMilestoneForm`
- Added `DatePickerFormatted` from EASI repository
- Added `MilestoneStatusTag`
- Added localstorage and default open state for table actions component
- Moved `ActionMenu` to it's own file
- Added hook `useFormatMTOCategories` extracted from previous code for reuse


## How to test this change

1.  Seed db
2. Navigate to MTO matrix
3. Click to edit milestone in the table row
4. Verify the `Saved changes` accurate displayed unsaved changes
5. Verify the leave modal appears to confirm when there are unsaved changes
6. Verify submission, removal, success, and error states
7. Try to break

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
